meteor/sofie-server/data-access/repositories/mongo/mongo-rundown-playlist-repository.ts

```diff
@@ -0,0 +1,76 @@
+import { RundownRepository } from '../interfaces/rundown-repository'
+import { BaseMongoRepository } from './base-mongo-repository'
+import { Rundown } from '../../../model/entities/rundown'
+import { MongoDatabase } from './mongo-database'
+import { MongoEntityConverter, MongoRundownPlaylist } from './mongo-entity-converter'
+import { BasicRundown } from '../../../model/entities/basic-rundown'
+
+const RUNDOWN_PLAYLIST_COLLECTION_NAME: string = 'rundownPlaylists'
+
+export class MongoRundownPlaylistRepository extends BaseMongoRepository implements RundownRepository {
+	constructor(
+		mongoDatabase: MongoDatabase,
+		mongoEntityConverter: MongoEntityConverter,
+		private rundownRepository: RundownRepository
+	) {
+		super(mongoDatabase, mongoEntityConverter)
+	}
+
+	protected getCollectionName(): string {
+		return RUNDOWN_PLAYLIST_COLLECTION_NAME
+	}
+
+	async getBasicRundowns(): Promise<BasicRundown[]> {
+		this.assertDatabaseConnection(this.getBasicRundowns.name)
+		const basicRundowns: BasicRundown[] = await this.rundownRepository.getBasicRundowns()
+
+		const playlists: MongoRundownPlaylist[] = (await this.getCollection()
+			.find()
+			.project({})
+			.toArray()) as unknown as MongoRundownPlaylist[]
+
+		const activePlaylist: MongoRundownPlaylist | undefined = playlists.find((playlist) => playlist.activationId)
+		if (!activePlaylist) {
+			return basicRundowns
+		}
+
+		const basicRundown: BasicRundown | undefined = basicRundowns.find(
+			(basicRundown) => basicRundown.name === activePlaylist.externalId
+		)
+		if (!basicRundown) {
+			return basicRundowns
+		}
+
+		Object.assign(
+			basicRundown,
+			new BasicRundown(basicRundown.id, basicRundown.name, true, basicRundown.getLastTimeModified())
+		)
+
+		return basicRundowns
+	}
+
+	async getRundown(rundownId: string): Promise<Rundown> {
+		this.assertDatabaseConnection(this.getRundown.name)
+		const rundown: Rundown = await this.rundownRepository.getRundown(rundownId)
+		const isRundownActive = await this.isRundownActive(rundown)
+
+		const activatedRundown: BasicRundown = new BasicRundown(
+			rundown.id,
+			rundown.name,
+			isRundownActive,
+			rundown.getLastTimeModified()
+		)
+		Object.assign(rundown, activatedRundown)
+		return rundown
+	}
+
+	private async isRundownActive(rundown: Rundown) {
+		const cursor = this.getCollection().find({ externalId: rundown.name }).project({ _id: 0, activationId: 1 })
+		const activationStatus = (await cursor.toArray())[0]
+		return Object.entries(activationStatus).length > 0
```

Comment:
This is causing an error for me. There's no guarantee that `cursor.toArray()` returns at least one document. If you want to know how many documents with a specified  `externalId` have any value in `activationId`, I think you should do something like `this.getCollection().countDocuments({ externalId: rundown.name, activationId: { $exists: true } })`

Reply:
The idea was to check whether or not an activationId was set, but though i don't get the error i do see an issue. However, this will change with what i am working on currently, since the playlist-layer should become superfluous.